src/integrationTest/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerEditRecordListingIT.java

```diff
@@ -0,0 +1,301 @@
+        final CaseData caseData = CaseData.builder()
+            .build();
+
+        final User user = new User(
+            TEST_AUTHORIZATION_TOKEN,
+            UserDetails.builder()
+                .roles(List.of("caseworker-st_cic", "caseworker-sptribs-systemupdate"))
+                .build()
+        );
+
+        stubForIdamDetails(TEST_AUTHORIZATION_TOKEN, CASEWORKER_USER_ID, ST_CIC_CASEWORKER);
+        when(authTokenGenerator.generate()).thenReturn(TEST_SERVICE_AUTH_TOKEN);
+        when(idamService.retrieveSystemUpdateUserDetails()).thenReturn(user);
+        when(judicialClient.getUserProfiles(
+            eq(TEST_SERVICE_AUTH_TOKEN),
+            eq(TEST_AUTHORIZATION_TOKEN),
+            eq(ACCEPT_VALUE),
+            eq(JudicialUsersRequest.builder().ccdServiceName(ST_CIC_JURISDICTION).build())
+        )).thenReturn(getUserProfiles());
+
+        String response = mockMvc.perform(post(ABOUT_TO_START_URL)
+                .contentType(APPLICATION_JSON)
+                .header(SERVICE_AUTHORIZATION, TEST_AUTHORIZATION_TOKEN)
+                .header(AUTHORIZATION, TEST_AUTHORIZATION_TOKEN)
+                .content(objectMapper.writeValueAsString(
+                    callbackRequest(
+                        caseData,
+                        CASEWORKER_EDIT_RECORD_LISTING)))
+                .accept(APPLICATION_JSON))
+            .andExpect(
+                status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        assertThatJson(response)
+            .when(IGNORING_EXTRA_FIELDS)
+            .isEqualTo(json(expectedResponse(CASEWORKER_EDIT_RECORD_LISTING_ABOUT_TO_START_RESPONSE)));
+    }
+
+    @Test
+    void shouldReturnSuccessMessageOnSubmitted() throws Exception {
+        final CaseData caseData = CaseData.builder()
+            .cicCase(CicCase.builder()
+                .hearingNotificationParties(emptySet())
+                .build())
+            .build();
+
+        String response = mockMvc.perform(post(SUBMITTED_URL)
+            .contentType(APPLICATION_JSON)
+            .header(SERVICE_AUTHORIZATION, TEST_AUTHORIZATION_TOKEN)
+            .header(AUTHORIZATION, TEST_AUTHORIZATION_TOKEN)
+            .content(objectMapper.writeValueAsString(
+                callbackRequest(
+                    caseData,
+                    CASEWORKER_EDIT_RECORD_LISTING)))
+            .accept(APPLICATION_JSON))
+            .andExpect(
+                status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        assertThatJson(response)
+            .inPath(CONFIRMATION_HEADER)
+            .isString()
+            .contains("""
+                # Listing record updated\s
+                ##  If any changes are made to this hearing,  remember to make those changes in this listing record.\s
+                ## A notification has been sent to""");
```

Comment:
Are we able to test sending to all parties? There is an example in one of the integration tests I have merged.

Reply:
Thanks, it's done.